packages/@react-aria/overlays/src/useOverlayPosition.ts

```diff
@@ -124,11 +124,23 @@ export function useOverlayPosition(props: AriaPositionProps): PositionAria {
     arrowSize
   ];
 
+  // Upon opening a overlay, record the current visual viewport scale so we can freeze the overlay styles
+  let lastScale = useRef(visualViewport?.scale);
+  useEffect(() => {
+    if (isOpen) {
+      lastScale.current = visualViewport?.scale;
+    }
+  }, [isOpen]);
+
```

Comment:
Some weirdness that I'm still investigating:
- In the storybook when opened outside the iframe, adding this "freeze" to the overlay styles makes the overlay positioning break as you pinch to zoom. This seems to be due to some storybook weirdness where the body of the page doesn't actually extend to the bottom of the page, instead being truncated so it is no longer the same height as it was prior to the pinch zoom operation. I'm not entirely sure why this is, haven't had much luck reproducing outside our storybook nor do I see anything in the styles themselves that would seem to cause this.
If you test pinch in the built docs for ComboBox, you'll notice that it works just fine

- The overlay will still reposition itself if you've pinch zoom in first, open the overlay, and then scroll around (this scrolling isn't the same as the standard scroll event, it is specifically for the visual viewport since you've zoomed in). This is due to the following: https://github.com/adobe/react-spectrum/blob/56147d528a25bd5849bf8e0e60e7f06a8355cf82/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L175-L187 which was added to fix overlay positioning when opening an overlay in iOS aka we recalculate the positioning on visual viewport scroll. At a bit of a loss of how best to differentiate the two scroll scenarios